flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/ScalarOperatorsTest.scala

```diff
@@ -240,3 +243,266 @@ class ScalarOperatorsTest extends ScalarOperatorsTestBase {
     testSqlApi("uuid() = cast(f22 as timestamp_ltz)", "null")
   }
 }
+
```

Comment:
Thank you @tsreaper, It would be great to add new tests using Java instead of scala.
Please check `CastFunctionITCase` and maybe you can use something similar for these tests.